[docs] Update Sentry docs for EAS Update compatibility (#19753)
expo/expo: docs/pages/guides/using-sentry.mdx

- Your `dist` will automatically be set to Sentry's expected value: `${buildNumber}` (iOS) or `${versionCode}` (Android).
- The configuration for build time sourcemaps comes from the **ios/sentry.properties** and **android/sentry.properties** files. For more information, refer to [Sentry's documentation](https://docs.sentry.io/clients/java/config/#configuration-via-properties-file). Manual configuration is only required for bare projects, the [sentry-expo config plugin handles it otherwise](#add-the-config-plugin).
- Configuration for `expo publish` and `npx expo export` for projects is done via **app.json**, whether using bare workflow or not.

Skipping or misconfiguring either of these can lead to invalid sourcemaps, and you won't see human-readable stacktraces in your errors.

### Uploading sourcemaps for updates

If you're using EAS Update, or if you're self-hosting your updates (this means you run `npx expo export` manually), you need to take the following steps to upload the sourcemaps for your update to Sentry:

- Running `eas update` or `npx expo export` will generate a **dist** folder in your project root, which contains your JavaScript bundles and sourcemaps.

> - Next, you can use the Sentry CLI to upload your bundles and sourcemaps:

Author: In a future iteration, maybe, but this initial iteration just allows you to use `sentry-cli` directly, and makes sure `sentry-expo` uses the correct release and dist values to match that out of the box.

<Tabs>
<Tab label="iOS">
<Terminal cmd={[
  "$ node_modules/@sentry/cli/bin/sentry-cli releases \\",

>   "    files <release name> \\",

Author: Yes, if you run `eas update ... --json` then the updateId for both platforms will be output that you could automatically read and use when calling `sentry-cli` to pass in as dist (or release). We will default to passing updateId in to `dist` though, so if you want to use it as `release` value then you might need to tweak the `Sentry.init` call a bit to make that work properly.

>   "    files <release name> \\",

Reviewer: > We will default to passing updateId in to `dist` though, so if you want to use it as `release` value then you might need to tweak the `Sentry.init` call a bit to make that work properly.

Ah cool. Ties into #297 which I just noticed you commented on this morning 😀

- Your `release` will automatically be set to Sentry's expected value- `${bundleIdentifier}@${version}+${buildNumber}` (iOS) or `${androidPackage}@${version}+${versionCode}` (Android).
- Your `dist` will automatically be set to Sentry's expected value: `${buildNumber}` (iOS) or `${versionCode}` (Android).

> - The configuration for build time sourcemaps comes from the **ios/sentry.properties** and **android/sentry.properties** files. For more information, refer to [Sentry's documentation](https://docs.sentry.io/clients/java/config/#configuration-via-properties-file). Manual configuration is only required for bare projects, the [sentry-expo config plugin handles it otherwise](#add-the-config-plugin).

Reviewer 2: ```suggestion
- The configuration for build time sourcemaps comes from the **android/sentry.properties** and **ios/sentry.properties** files. For more information, see [Sentry's documentation](https://docs.sentry.io/clients/java/config/#configuration-via-properties-file). Manual configuration is only required for bare projects, the [sentry-expo config plugin handles it otherwise](#add-the-config-plugin).
```

[Suggestion to prefer Android before iOS as per writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#referencing-android-ios-and-web).

> - Configuration for `expo publish` and `npx expo export` for projects is done via **app.json**, whether using bare workflow or not.

Reviewer 2: Not sure about this but do we still need reference global Expo CLI commands? And if yes, I think it will be suffixed by `expo-cli`?

Reviewer 2: 🤔 You are right we don't specify using `expo-cli` as a command and we don't have a standard as of now. I think this can be marked as resolved and we can circle back later to update commands if required in future.

Author: Sounds good. Thanks for your sharp eye though!

Skipping or misconfiguring either of these can lead to invalid sourcemaps, and you won't see human-readable stacktraces in your errors.

### Uploading sourcemaps for updates

> This requires SDK 47, with `sentry-expo@6.0.0` or above and `expo-updates@0.15.5` or above.

If you're using EAS Update, or if you're self-hosting your updates (this means you run `npx expo export` manually), you need to take the following steps to upload the sourcemaps for your update to Sentry:

> - Run `eas update`. This will generate a **dist** folder in your project root, which contains your JavaScript bundles and sourcemaps. This command will also output the 'Android update ID' and 'iOS update ID' that we'll need in the next step.

Reviewer: > This command will also output the 'Android update ID' and 'iOS update ID' that we'll need in the next step.

Are there any plans to make this smoother? We run `eas update` on CI, so unless we wanna parse the stdout of `eas update` and fish this out, we cannot really use this.

---

Ideally this entire process would be wrapped up in a `require('sentry-expo/eas-update')` or some such, but I assume there's a reason that's not done 🙂

> - Run `eas update`. This will generate a **dist** folder in your project root, which contains your JavaScript bundles and sourcemaps. This command will also output the 'Android update ID' and  'iOS update ID' that we'll need in the next step.

Reviewer 2: ```suggestion
- Run `eas update`. This will generate a **dist** folder in your project root, which contains your JavaScript bundles and sourcemaps. This command will also output the 'Android update ID' and 'iOS update ID' that we'll need in the next step.
```

> - Copy or rename the bundle names in the **dist/bundles** folder to match **main.jsbundle** (iOS) or **index.android.bundle** (Android).

Reviewer 2: ```suggestion
- Copy or rename the bundle names in the **dist/bundles** folder to match **index.android.bundle** (Android) or **main.jsbundle** (iOS).
```
